### ISSUE TITLE
"Just Enough Dimensions" mod incompatibility

### ISSUE THREAD

**Issue by xkforce** [jbredwards/Nether-API#3]
Whether the mod modifies the terrain or not, netherapi expects its worldproviders to be used in the nether and end. While troubleshooting a crash that occurred after two worlds are created without a restart of the game, I realized that justenoughdimensions and netherapi were conflicting with one another. Specifically the JED worldprovider needs to be used for the End because some JED functionality necessary for the pack to function properly in that dimension requires JED's provider. The nether worldprovider does not need to be JED's which is why only the end worldprovider was an issue in this particular case. Currently I do not have a workaround for this that would allow the functionality needed for the End to be compatible with netherapi.

**Comment by jbredwards** [comment]
I'm changing the title of this issue to "Just Enough Dimensions mod incompatibility" to reflect the actual problem, since it wouldn't be too hard to add compatibility in the future :)

The inability to toggle this mod's world providers is by design. They're what makes it possible to add custom biomes, structures, fog color, music, ambient effects (particles, sounds), and edit structures, which is all kinda the whole point of the mod :P

**Comment by xkforce** [comment]
JED requires that its worldprovider be used for a whole swath of its features to work in that dimension for the same reasons nether api does.

**Comment by jbredwards** [comment]
> JED requires that its worldprovider be used

Not necessarily. All that compatibility requires is for the active world provider to call JED functions and use its settings while that mod is present. I'll add that functionality in the future (probably in the next update or so)

**Comment by CalaMariGold** [comment]
Possibly related crash that occurred during world creation: https://mclo.gs/BIBAPt3

**Comment by jbredwards** [comment]
> Possibly related crash that occurred during world creation: https://mclo.gs/BIBAPt3

@CalaMariGold This looks like the result of removing the nether or end entirely (and thus isn't exclusively related to JED), so can you move this to a new issue?

**Comment by CalaMariGold** [comment]
I haven't removed either dimension 🤔

**Comment by CalaMariGold** [comment]
I do override it with JED tho

```
{
    "dimensions": [
        {
            "dim": -1,
            "override": true,
            "dimensiontype": {
                "id": -1,
                "name": "Nether (JED)",
                "suffix": "_dim-1",
                "keeploaded": false,
                "worldprovider": "WorldProviderHellJED",
                "worldinfo_onetime": {}
            },
            "jed": {
                "CanRespawnHere": true,
                "CanSleepHere": "deny"
            }
        }
    ]
}
```

**Comment by jbredwards** [comment]
@CalaMariGold Can you send the full log (not just the crash-log)?

**Comment by CalaMariGold** [comment]
https://mclo.gs/7CxfXh7

**Comment by CalaMariGold** [comment]
Hi hi, any updates to this issue?

**Comment by jbredwards** [comment]
> Hi hi, any updates to this issue?

It's hard working on multiple mods while also being a full-time college student, so I'll add support during winter break :)

**Comment by CalaMariGold** [comment]
Absolutely no worries at all ❤️ Take your time!

**Comment by jbredwards** [comment]
Fixed in v1.4.0! No changes need to be made to existing configs (including the "worldprovider" setting).

**Comment by jbredwards** [comment]
@Terra-B-Welch use the latest version of JED
